optimum/onnx/configuration.py

```diff
@@ -339,6 +339,73 @@ def fill_with_past_key_values_(self, inputs_or_outputs: Mapping[str, Mapping[int
         for i in range(num_decoder_layers * num_pkv_per_layer):
             inputs_or_outputs[f"{name}_key_values_{i}"] = {0: "batch", 2: decoder_sequence}
 
+    def generate_dummy_inputs_onnxruntime(self, reference_model_inputs: Mapping[str, Any]) -> Mapping[str, Any]:
+        reference_model_inputs["encoder_attention_mask"] = reference_model_inputs.pop("attention_mask")
+        reference_model_inputs["input_ids"] = reference_model_inputs.pop("decoder_input_ids")
+        reference_model_inputs["encoder_hidden_states"] = reference_model_inputs.pop("encoder_outputs")[0]
+
+        return reference_model_inputs
+
+    @property
+    def values_override(self) -> Optional[Mapping[str, Any]]:
+        if hasattr(self._config, "use_cache"):
+            return {"use_cache": True}
+
+        return None
+
+
+class SpeechSeq2SeqEncoderOnnxConfig(OnnxConfig):
+    @property
+    def inputs(self) -> Mapping[str, Mapping[int, str]]:
+        return OrderedDict(
+            [
+                ("input_features", {0: "batch", 1: "feature_size", 2: "encoder_sequence"}),
+            ]
+        )
+
+    @property
+    def outputs(self) -> Mapping[str, Mapping[int, str]]:
+        return OrderedDict({"last_hidden_state": {0: "batch", 1: "encoder_sequence"}})
+
+    @property
+    def atol_for_validation(self) -> float:
+        return 1e-4
+
+
+class SpeechSeq2SeqDecoderOnnxConfig(DecoderOnnxConfig):
+    @property
+    def inputs(self) -> Mapping[str, Mapping[int, str]]:
+        common_inputs = OrderedDict(
+            [
+                ("input_ids", {0: "batch", 1: "past_decoder_sequence + sequence"}),
+                ("encoder_hidden_states", {0: "batch", 1: "encoder_sequence"}),
+            ]
+        )
+        if self.use_past:
+            self.fill_with_past_key_values_(common_inputs, direction="inputs")
+
+        return common_inputs
+
+    def generate_dummy_inputs(
```

Comment:
FYI this method will be superseded by @michaelbenayoun's ONNX PR in #403 

It's probably OK to keep things as they are for now and then refactor later IMO (unless there's objections from the team of course :))

Reply:
Refactored based on exporters